Examples/MAX32660/Flash/main.c

```diff
@@ -171,233 +78,225 @@ void FLC0_IRQHandler(void)
+int write_test_pattern()
 {
-    int retval;
-    uint32_t start_align, start_len, end_align, end_len, i;
-
-    MXC_ASSERT(buffer);
-
-    // Align start and end on page boundaries, calculate length of data to buffer
-    start_align = start - (start % MXC_FLASH_PAGE_SIZE);
-    start_len = (start % MXC_FLASH_PAGE_SIZE);
-    end_align = end - (end % MXC_FLASH_PAGE_SIZE);
-    end_len = MXC_FLASH_PAGE_SIZE - (end % MXC_FLASH_PAGE_SIZE);
-
-    // Make sure the length of buffer is sufficient
-    if ((length < start_len) || (length < end_len)) {
-        return E_BAD_PARAM;
+    int err;
+    // A flash address must be in the erased state before writing to it, because the
+    // flash controller can only write a 1 -> 0.
+    // See the microcontroller's User Guide for more details.
+    printf("Erasing page of flash at addr 0x%x...\n", TEST_ADDRESS);
+    err = MXC_FLC_PageErase(TEST_ADDRESS);
+    if (err) {
+        printf("Failed with error code %i\n", TEST_ADDRESS, err);
+        return err;
     }
 
-    // Start and end address are in the same page
-    if (start_align == end_align) {
-        if (length < (start_len + end_len)) {
-            return E_BAD_PARAM;
-        }
-
-        // Buffer first page data and last page data, erase and write
-        memcpy(buffer, (void *)start_align, start_len);
-        memcpy(&buffer[start_len], (void *)end, end_len);
-        retval = MXC_FLC_PageErase(start_align);
-        if (retval != E_NO_ERROR) {
-            return retval;
-        }
+    printf("Writing magic value 0x%x to address 0x%x...\n", MAGIC, TEST_ADDRESS);
+    err = MXC_FLC_Write32(TEST_ADDRESS, MAGIC);
+    if (err) {
+        printf("Failed to write magic value to 0x%x with error code %i!\n", TEST_ADDRESS, err);
+        return err;
+    }
 
-        retval = MXC_FLC_Write(start_align, start_len, buffer);
-        if (retval != E_NO_ERROR) {
-            return retval;
+    printf("Writing test pattern...\n");
+    for (uint32_t addr = TEST_ADDRESS + 4; addr < TEST_ADDRESS + MXC_FLASH_PAGE_SIZE; addr += 4) {
+        /*  
+            A single flash page is organized into 4096 128-bit "words", but is still 
+            byte-addressible.  Increment the address by 4 bytes to write in 32-bit 
+            chunks.  
+            
+            The Flash Controller also only supports 128-bit writes. The driver
+            function below handles the technicalities of inserting the 32-bit value 
+            into a 128-bit word without modifying the rest of the word.
+        */
+        err = MXC_FLC_Write32(addr, TEST_VALUE);
+        if (err) {
+            printf("Failed write on address 0x%x with error code %i\n", addr, err);
+            return err;
         }
-        retval = MXC_FLC_Write(end, end_len, &buffer[start_len]);
-        if (retval != E_NO_ERROR) {
-            return retval;
-        }
-
-        return E_NO_ERROR;
     }
+    printf("Done!\n");
```

Comment:
Should the write be validated here as well?

Reply:
Yes it makes sense to validate again here.  Added in https://github.com/Analog-Devices-MSDK/msdk/pull/357/commits/6c5880231020a53832db31a2eda14b4074570797